src/NServiceBus.Core/Unicast/Messages/MessageMetadataRegistry.cs

```diff
@@ -48,12 +48,21 @@ public MessageMetadata GetMessageMetadata(string messageTypeIdentifier)
         {
             Guard.AgainstNullAndEmpty(nameof(messageTypeIdentifier), messageTypeIdentifier);
 
-            var messageType = Type.GetType(messageTypeIdentifier, false);
+            var messageType = GetType(messageTypeIdentifier);
 
             if (messageType == null)
             {
                 Logger.DebugFormat("Message type: '{0}' could not be determined by a 'Type.GetType', scanning known messages for a match", messageTypeIdentifier);
-                return messages.Values.FirstOrDefault(m => m.MessageType.FullName == messageTypeIdentifier);
+
+                foreach (var item in messages.Values)
+                {
+                    if (item.MessageType.FullName == messageTypeIdentifier)
```

Comment:
Is the assmbly version differs btw the sender and receiver we might not be able to "create" the type so that code is there to find the message type based on a perfect match on fullname. Eg. the SC plugins use this since the message are compile into the plugin and also into the SC dll it so the assemblies will differ. Does this make any sense?